opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -16,82 +16,138 @@
 from collections import OrderedDict
 from logging import getLogger
 from math import inf
+from threading import Lock
+from typing import Generic, Optional, Sequence, TypeVar
 
-from opentelemetry._metrics.instrument import _Monotonic
+from opentelemetry.sdk._metrics.measurement import Measurement
+from opentelemetry.sdk._metrics.point import Gauge, Histogram, PointT, Sum
 from opentelemetry.util._time import _time_ns
 
+# FIXME this is being copied directly from
+# opentelemetry.proto.metrics.v1.metrics_pb2. The only reason for doing so is
+# to avoid havinv protobuf as a indirect dependency in the SDK. This
+# duplication of code is not ideal.
+
+AGGREGATION_TEMPORALITY_UNSPECIFIED = 0
+AGGREGATION_TEMPORALITY_DELTA = 1
+AGGREGATION_TEMPORALITY_CUMULATIVE = 2
+
+_PointVarT = TypeVar("_PointVarT", bound=PointT)
+
 _logger = getLogger(__name__)
 
 
-class Aggregation(ABC):
+class Aggregation(ABC, Generic[_PointVarT]):
+    def __init__(self, is_monotonic: bool):
+        self._value = None
+        self._is_monotonic = is_monotonic
+        self._lock = Lock()
+
+    @property
+    def is_monotonic(self):
+        return self._is_monotonic
+
     @property
     def value(self):
-        return self._value  # pylint: disable=no-member
+        return self._value
 
     @abstractmethod
-    def aggregate(self, value):
+    def aggregate(self, measurement: Measurement) -> None:
         pass
 
     @abstractmethod
-    def make_point_and_reset(self):
-        """
-        Atomically return a point for the current value of the metric and reset the internal state.
-        """
-
+    def _collect(self) -> Optional[_PointVarT]:
```

Comment:
AFAIK `AsynchronousSumAggregation` does not reset anything, so `make_point_and_reset` is misleading. Metrics is pretty much 2 sequences of method calls, `process` and `collect`. It is a long sequence in both cases and it gets very confusing if we start using different method names, this way you can know which sequence each method belongs to. I can make it public instead of private :v:

Reply:
I'm fine with `collect()` then, but can we add back the comment about resetting? The `AsynchronousSumAggregation` would still reset. It can throw away the previous measurement

Reply:
All right, adding the docstring back :+1: